vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -2957,6 +2957,22 @@ func TestAccResourceVSphereVirtualMachine_importClone(t *testing.T) {
 	})
 }
 
+// testAccResourceVSphereVirtualMachineReadVappChildResourcePool
+// TestAccResourceVSphereVirtualMachine_importClone
+func TestAccResourceVSphereVirtualMachine_readVappChildResourcePool(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccResourceVSphereVirtualMachinePreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceVSphereVirtualMachineReadVappChildResourcePool(),
+			},
```

Comment:
It probably isn't really necessary, but it might be good to throw in a testAccResourceVSphereVirtualMachineCheckExists check here just to make sure the VM gets created as expected.